Integration of filter/sort feature into modals and card view

AaronDzf opened the pull request:
Filter & sort features now will update the cards under a task list rather than in a separate window

- Removed iframes implementation, changed into modal to be consistent with codebase
- filterList.php no longer in use

- Test cases:
• Filter/sort applies to individual task lists
• Filter/sort applies within board
• Filter/sort features work regardless of user permission (e.g. Edit or
View permission)

Rolo123y committed:
transforming from iframe to default modal implementation.

AaronDzf committed:
Integration of filter/sort features into modals and the card view
- Filter & Sort features now display in the cards under a task list
- Removed iframes implementation
- Test cases:
• Filter/sort applies to individual task lists
• Filter/sort applies within board
• Filter/sort features work regardless of user permission (e.g. Edit or
View permission)

Rolo123y commented on filterList.php:
```diff
@@ -90,7 +90,6 @@
 
 <!-- Form to take in user inputs -->
 <form action="filterList.php?id=<?php echo $_GET['id']; ?>" method="POST"> 
```
This page is no longer required if we are going with the new filtering implementation in listView.php. In the current implementation, the content in filterList.php is transferred to listView.php much like the viewTask content which was also moved into listView.php after the removal of iframe.

AaronDzf replied:
For sure but we may want to keep this file up until we finalize these changes just so we don't fully remove them yet.

Rolo123y commented on listView.php:
```diff
@@ -376,22 +441,66 @@
+          <!-- <iframe sandbox="allow-top-navigation allow-scripts allow-forms" class="embed-responsive-item" id="filterListFrame" style="border:0; width:558px; height:700px;" src="listView.php"></iframe> -->
+          <form action="listView.php?board=<?php echo $board['boardID']; ?>" method="POST"> 
+              <input type="hidden" name="filterListID" value="<?php echo $list['listID']; ?>"> 
+              <br>
+              <input type="text" name="filt_name" value="" class="form-control" placeholder="Task Title" maxlength="255"/>
+              <br>
+              <input type="number" name="filt_effort" value="" class="form-control" placeholder="Importance" min = "0" max = "10" step="1"/>
+              <br>
+              <input type="text" name="filt_work" value="" class="form-control" placeholder="Type of Work" maxlength="255"/>
+              <br>
+              <input type="datetime-local" name="filt_date" value="unchanged" class="form-control" placeholder="Due Date"/>
+              <br>
+              <select name="sortTask">
```
The < br> can come right after the input to make this easier to read. 

<input type="" ... ></input>< br>

Rolo123y replied:
The inputs should come with a label rather than using placeholders. This would be better for the user to know what each input is. The 'Due Date' placeholder does not show in the current UI so the user does not know what to input there.

![Capture](https://user-images.githubusercontent.com/52054023/101232386-7369eb80-367f-11eb-8387-b6dde7be7403.PNG)

AaronDzf replied:
Will add labels under the inputs similar to the add new task modal

Rolo123y replied:
Looks good now. It is easy to know what each input is.
![Capture](https://user-images.githubusercontent.com/52054023/101233160-95b23800-3684-11eb-8d80-a5d63c6be8fa.PNG)

Rolo123y commented on listView.php:
```diff
@@ -256,23 +257,87 @@
       ".$list['listTitle']."  
     </div>
   </div>
 
 
```
remove empty lines.

Rolo123y replied:
Also, the </div> components can be indented to match the corresponding starting <div>.  For example, line 261 does not match line 259.

AaronDzf replied:
Noted! Thanks

Rolo123y reviewed (changes requested):
- Logging in as user: wmmeyer; 342Group8!?
- testing using BoardID = 2

**Tested the UI/UX with the following tests:**
- [X] Clicking on the list title to see the unchanged filtering pop-up
- [X] Clicking 'filter' button should exit the pop-up
- [X] The list should show only the filtered tasks on the Board

**Tested the functionality of the change with the following tests:**
-  For the same Board and the same list
  
- [x]  Filtered by task title; The complete task title must be inputted and outputs the task in the list once 'filter' is clicked
- [X] Filtered by importance of 1; Only task 'Monday Task' is shown and aligns with the database values
- [X] Filtered by typeofWork 'Monday'; Only task 'Monday Task' is shown and aligns with the database values
- [X] Filtered by typeofWork 'Monday'; The output in the list is as expected and aligns with the database values
- [X] Sorting; The output in the list is as expected and aligns with the database values when sorting by each sorting method in Ascending or Descending order.

AaronDzf committed:
Added the following changes:
•Added labels under each input in the modal for user clarity
•Added headers in the modal
•Removed blank lines
•Added comments to the output statements of tasks in a task list
•General indentation cleanup

AaronDzf commented:
Scroll bar looks good!
I also looked at the archive page and that hasn't implemented scrolling, maybe we should take a look at that?
Otherwise, no issues on my end.

Tested the feature for:
- [X] Scroll bar does not appear if the number of tasks does not exceed the height of the task list card
- [X] Scroll bar appears correctly when number of tasks exceeds height of the task list card
- [X] Scroll bar disappears correctly if tasks are archived and number of tasks does not exceed height of the task list card
- [X] Scroll bar functions correctly
